contracts/btc-staking/src/validation/mod.rs

```diff
@@ -75,3 +83,165 @@ pub fn verify_new_fp(new_fp: &NewFinalityProvider) -> Result<(), ContractError>
+        &staker_pk,
+        active_delegation.unbonding_time as u16,
+    )?;
+
+    // TODO: Verify proof of possession
+
+    /*
+        verify staker signature against slashing path of the staking tx script
+    */
+
+    // get the slashing path script
+    let staking_output = &staking_tx.output[active_delegation.staking_output_idx as usize];
+    let staking_time = (active_delegation.end_height - active_delegation.start_height) as u16;
+    let babylon_script_paths = babylon_btcstaking::scripts_utils::BabylonScriptPaths::new(
+        &staker_pk,
+        &fp_pks,
+        &cov_pks,
+        params.covenant_quorum as usize,
+        staking_time,
+    )?;
+    let slashing_path_script = babylon_script_paths.slashing_path_script;
+
+    // get the staker's signature on the slashing tx
+    let staker_sig =
+        k256::schnorr::Signature::try_from(active_delegation.delegator_slashing_sig.as_slice())
+            .map_err(|e| ContractError::SecP256K1Error(e.to_string()))?;
+
+    // Verify the signature
+    babylon_btcstaking::sig_verify::verify_transaction_sig_with_output(
+        &slashing_tx,
+        staking_output,
+        slashing_path_script.as_script(),
+        &staker_pk,
+        &staker_sig,
+    )
+    .map_err(|e| ContractError::SecP256K1Error(e.to_string()))?;
+
+    // TODO: verify covenant signatures
+
+    // TODO: Check unbonding time (staking time from unbonding tx) is larger than min unbonding time
+    // which is larger value from:
+    // - MinUnbondingTime
+    // - CheckpointFinalizationTimeout
+
+    // At this point, we know that unbonding time in request:
+    // - is larger than min unbonding time
+    // - is smaller than math.MaxUint16 (due to check in req.ValidateBasic())
+
+    /*
+        TODO: Early unbonding logic
+    */
+
+    // TODO: Deserialize provided transactions
+
+    // TODO: Check that the unbonding tx input is pointing to staking tx
+
+    // TODO: Check that staking tx output index matches unbonding tx output index
+
+    // TODO: Build unbonding info
+
+    // TODO: Get unbonding output index
+
+    // TODO: Check that slashing tx and unbonding tx are valid and consistent
+
+    // TODO: Check staker signature against slashing path of the unbonding tx
+
+    // TODO: Verify covenant signatures over unbonding slashing tx
+
+    // TODO: Check unbonding tx fees against staking tx
+    // - Fee is greater than 0.
+    // - Unbonding output value is at least `MinUnbondingValue` percentage of staking output value.
+
+    Ok(())
+}
+
+/// verify_active_delegation is a placeholder for the full validation logic
+///
+/// It is marked with `#[cfg(not(feature = "full-validation"))]` so that it
+/// is not included in the build if the `full-validation` feature is enabled.
+#[cfg(not(feature = "full-validation"))]
+pub fn verify_active_delegation(
+    _params: &Params,
+    _active_delegation: &ActiveBtcDelegation,
+    _staking_tx: &Transaction,
+) -> Result<(), ContractError> {
+    Ok(())
+}
```

Comment:
Nit: An alternative that is perhaps cleaner, is instead of having two functions, have just one and put the `full-vaildation` feature in the body.

The warning about missing parameters can be avoided by doing something like

`let _ = (params, active_delegation, staking_tx)` in an extra `not` section.

Reply:
good ideas!